spring-integration-ip/src/main/java/org/springframework/integration/ip/tcp/connection/TcpNioServerConnectionFactory.java

```diff
@@ -123,6 +124,7 @@ public void run() {
 			if (logger.isInfoEnabled()) {
 				logger.info(this + " Listening");
 			}
+			publishServerListeningEvent(getPort());
```

Comment:
Looks like I see an issue with `TcpNioConnectionTests.int3453RaceTest`.
Previously we waited for the `TcpConnectionOpenEvent`, but now we have a new one `TcpConnectionServerListeningEvent`, which definitely is emitted before `connection.publishConnectionOpenEvent();`.
The
```
if (event instanceof TcpConnectionOpenEvent) {
	connectionLatch.countDown();
}
```
fixes the early exit from the `factory` setup.